packages/x-data-grid-premium/src/components/promptControl/GridToolbarPromptControl.tsx

```diff
@@ -0,0 +1,276 @@
+  const [error, setError] = React.useState<string | null>(null);
+  const [isRecording, setRecording] = React.useState(false);
+  const [query, setQuery] = React.useState('');
+
+  const classes = useUtilityClasses(rootProps, isRecording);
+  const examplesFromData = React.useMemo(
+    () => (allowDataSampling ? sampleData(apiRef) : undefined),
+    [apiRef, allowDataSampling],
+  );
+
+  const processPrompt = React.useCallback(() => {
+    const context = generateContext(apiRef, examplesFromData);
+    const columnsByField = gridColumnLookupSelector(apiRef);
+
+    setLoading(true);
+    setError(null);
+    apiRef.current.setState((state) => ({ ...state, rows: { ...state.rows, loading: true } }));
+
+    onPrompt(context, query)
+      .then((result) => {
+        const interestColumns = [] as string[];
+
+        apiRef.current.setFilterModel({
+          items: result.filters.map((filter, index) => {
+            const item = {
+              id: index,
+              field: filter.column,
+              operator: filter.operator,
+              value: filter.value,
+            };
+
+            const column = columnsByField[filter.column];
+            if (column.type === 'singleSelect') {
+              const options = getValueOptions(column as GridSingleSelectColDef) ?? [];
+              const found = options.find(
+                (option) => typeof option === 'object' && option.label === filter.value,
+              );
+              if (found) {
+                item.value = (found as any).value;
+              }
+            }
+
+            return item;
+          }),
+          logicOperator: (result.filterOperator as GridLogicOperator) ?? GridLogicOperator.And,
+          quickFilterValues: [],
+        });
+
+        apiRef.current.setRowGroupingModel(result.grouping.map((g) => g.column));
+        apiRef.current.setAggregationModel(result.aggregation);
+
+        apiRef.current.setSortModel(
+          result.sorting.map((s) => ({ field: s.column, sort: s.direction })),
+        );
+
+        const rows = getVisibleRows(apiRef, rootProps);
+        const selectedRowIds =
+          result.select === -1
+            ? []
+            : rows.rows.slice(0, result.select).map((r) => {
+                return apiRef.current.getRowId(r);
+              });
+        apiRef.current.setRowSelectionModel(selectedRowIds);
+
+        const columns = apiRef.current.getAllColumns();
+        const targetIndex =
+          Number(columns.find((c) => c.field === GRID_CHECKBOX_SELECTION_FIELD) !== undefined) +
+          Number(result.grouping.length);
+
+        interestColumns.push(...Object.keys(result.aggregation));
+        interestColumns.push(...result.filters.map((f) => f.column));
+        interestColumns.reverse().forEach((c) => apiRef.current.setColumnIndex(c, targetIndex));
+      })
+      .catch((_) => {
+        setError(apiRef.current.getLocaleText('toolbarPromptControlErrorMessage'));
```

Comment:
I think it'd be useful to allow a custom error being thrown from `onPrompt` function. `toolbarPromptControlErrorMessage` could be used as fallback.

For example, `mockPromptResolver` can throw a message like:
```md
The query you posted is not registered with the mock prompt resolver.
Please try one of the above listed queries or use an actual API
```
Wdyt?

Reply:
I like this. Added it as a follow up task